app/assets/javascripts/templates/admin/modals/business_address_info.html.haml

```diff
@@ -0,0 +1,7 @@
+%div
+  .margin-bottom-30
+    %p
+      = I18n.t('js.admin.modals.business_address_info.message')
+
+  .text-center
+    %input.button.red.icon-plus{ type: 'button', value: I18n.t('js.admin.modals.got_it'), ng: { click: 'close()' } }
```

Comment:
The I18n setup is a bit confusing. We have a Ruby method called `I18n.t` and we have a Javascript method called `I18n.t`. The Javascript method is also called by the AngularJS filter `t`.

The way you wrote it, you are calling the Ruby method. And since this is a Javascript template it's precompiled and the Ruby method is called only once with the default language. When a user changes the language, it doesn't have any effect. You can test this by changing the language as user and then look at the modal:

![Screenshot from 2021-09-13 07-46-19](https://user-images.githubusercontent.com/3524483/133003852-195cc328-314d-46d1-90f6-178952d9db3e.png)

We need to write some Javascript code here, which is not executed during precompile but executed in the browser. It can then translate with the user's language. The easiest way is to use the AngularJS filter:

```suggestion
      {{ "js.admin.modals.business_address_info.message" | t }}

  .text-center
    %input.button.red.icon-plus{ type: 'button', value: '{{ "js.admin.modals.got_it" | t }}', ng: { click: 'close()' } }
```

Now we get this:

![Screenshot from 2021-09-13 08-02-34](https://user-images.githubusercontent.com/3524483/133004147-fe53a5ad-304f-491e-868b-a66a9237644d.png)

The "Got it" button is translated. The rest of the text is not translated yet just because the translation is missing from the locale but that will come once people translate.

I tried to look for good examples but found that we have many places where this is done right and many places where this is done wrong:

```diff
$ git grep -w t --  app/assets/javascripts/templates/admin/
-app/assets/javascripts/templates/admin/columns_dropdown.html.haml:  %span{ :class => 'icon-reorder' }= "&nbsp; #{t('admin.columns')}".html_safe
-app/assets/javascripts/templates/admin/columns_dropdown.html.haml:      %input.fullwidth.red{ type: "button", :value => t('admin.column_save_as_default').html_safe, ng: { show: "!saved() && !saving", click: "saveColumnPreferences(action)"} }
-app/assets/javascripts/templates/admin/edit_address_dialog.html.haml:  %h2 {{ addressType === 'bill_address' ?  "#{t('admin.customers.index.edit_bill_address')}" : "#{t('admin.customers.index.edit_ship_address')}" }}
+app/assets/javascripts/templates/admin/edit_address_dialog.html.haml:      {{ 'admin.customers.index.required_fileds' | t }}
+app/assets/javascripts/templates/admin/edit_address_dialog.html.haml:          {{ 'first_name' | t }}
+app/assets/javascripts/templates/admin/edit_address_dialog.html.haml:          {{ 'last_name' | t }}
+app/assets/javascripts/templates/admin/edit_address_dialog.html.haml:          {{ 'address' | t }}
+app/assets/javascripts/templates/admin/edit_address_dialog.html.haml:          {{ 'address2' | t }}
+app/assets/javascripts/templates/admin/edit_address_dialog.html.haml:          {{ 'city' | t }}
+app/assets/javascripts/templates/admin/edit_address_dialog.html.haml:          {{ 'postcode' | t }}
+app/assets/javascripts/templates/admin/edit_address_dialog.html.haml:          {{ 'country' | t }}
+app/assets/javascripts/templates/admin/edit_address_dialog.html.haml:          placeholder: "{{ 'admin.customers.index.select_country' | t }}",
+app/assets/javascripts/templates/admin/edit_address_dialog.html.haml:          {{ 'state' | t }}
+app/assets/javascripts/templates/admin/edit_address_dialog.html.haml:          placeholder: "{{ 'admin.customers.index.select_state' | t }}",
+app/assets/javascripts/templates/admin/edit_address_dialog.html.haml:          {{ 'phone' | t }}
+app/assets/javascripts/templates/admin/edit_address_dialog.html.haml:      %input.button.red.icon-plus{ type: 'submit', value: t('admin.customers.index.update_address')}
-app/assets/javascripts/templates/admin/info_dialog.html.haml:      = t(:ok)
+app/assets/javascripts/templates/admin/links_dropdown.html.haml:    {{ 'admin.actions' | t }}
-app/assets/javascripts/templates/admin/modals/bulk_invoice.html.haml:  = t('js.admin.orders.index.compiling_invoices')
-app/assets/javascripts/templates/admin/modals/bulk_invoice.html.haml:  = t('js.admin.orders.index.please_wait')
-app/assets/javascripts/templates/admin/modals/bulk_invoice.html.haml:  = t('js.admin.orders.index.view_file')
-app/assets/javascripts/templates/admin/modals/business_address_info.html.haml:      = I18n.t('js.admin.modals.business_address_info.message')
-app/assets/javascripts/templates/admin/modals/business_address_info.html.haml:    %input.button.red.icon-plus{ type: 'button', value: I18n.t('js.admin.modals.got_it'), ng: { click: 'close()' } }
+app/assets/javascripts/templates/admin/modals/image_upload.html.haml:  %label{for: 'image-upload', class: 'button'} {{ 'admin.products.index.upload_an_image' | t }}
-app/assets/javascripts/templates/admin/modals/invite_manager.html.haml:      = t('js.admin.modals.invite_title')
-app/assets/javascripts/templates/admin/modals/invite_manager.html.haml:      = t('js.admin.modals.invite')
-app/assets/javascripts/templates/admin/modals/invite_manager.html.haml:      = t('js.admin.modals.close')
+app/assets/javascripts/templates/admin/modals/tag_rule_help.html.haml:      {{ 'js.admin.modals.tag_rule_help.title' | t }}
...
```

I'm not sure if it's worth fixing all the other places because it will all be rebuilt when we replace AngularJS and then the translation issue will be solved at the same time.